### PR TITLE
chore(sdk): refactor to store type information in simulator.json

### PR DESCRIPTION
**Pull request by Chriscbr**
Initial steps towards #4667.

When you compile to the `sim` target, info about the app's resources is stored in `simulator.json`. This PR updates the file to include metadata about each resource's type, such as the simulation's source code location. This eliminates an ugly factory class that we've had to update each time someone adds a new resource to the simulator. 🙌

Since the Wing Console and SDK are happily living together in our monorepo and packaged together as part of the CLI, I've also cleaned up some cross-library type information so there's just one fully-qualified name constant associated with each SDK resource instead of two (see libs/wingsdk/src/target-sim/schema-resources.ts for the removal).

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/target-sim/domain.ts [hunk 1]
```diff
@@ -16,4 +17,13 @@ export class Domain extends cloud.Domain {
...
   }
+
+  public toSimulator(): BaseResourceSchema {
+    return {
+      type: cloud.DOMAIN_FQN,
+      path: this.node.path,
+      props: {},
+      attrs: {},
+    };
+  }
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/target-sim/domain.ts:28]: Domain isn't currently supported in the sim, right?  Should this throw or something?

File: libs/wingsdk/test/core/__snapshots__/connections.test.ts.snap [hunk 2]
```diff
@@ -56,10 +56,88 @@ return class Handler {
...
         },
-        "type": "wingsdk.cloud.Function",
+        "type": "@winglang/sdk.cloud.Function",
       },
     ],
     "sdkVersion": "0.0.0",
+    "types": {
+      "@winglang/sdk.cloud.Api": {
+        "className": "Api",
+        "sourcePath": "<ABSOLUTE PATH>/api.inflight.js",
```
Comment (MarkMcCulloh) [inline libs/wingsdk/test/core/__snapshots__/connections.test.ts.snap:66]: Will this need to be resolvable across machines? If so it'd be useful for this to be a module-relative path instead. I think it's fine to iterate on this but just wanted to check to see how critical the portability of simulator.json is.

I guess we could eventually just plop an entire JSII manifest in here haha.

**Review by MarkMcCulloh — APPROVED** [review]

File: libs/wingsdk/test/core/__snapshots__/connections.test.ts.snap (hunk 2 above)
Comment (Chriscbr) [reply libs/wingsdk/test/core/__snapshots__/connections.test.ts.snap:66]: > Will this need to be resolvable across machines?

I think with the changes we have planned to allow the simulator running across multiple processes, there's still an assumption of a "master" process (the one which started the simulation) that is responsible for initializing all of the resources -- at the moment I think that's the only process that needs to resolve these paths. But it's a good question, I think we we'll revise this format many times in the future

File: libs/wingsdk/src/target-sim/domain.ts (hunk 1 above)
Comment (Chriscbr) [reply libs/wingsdk/src/target-sim/domain.ts:28]: I wasn't sure - I see we have an open issue for adding support [here](https://github.com/winglang/wing/issues/4556), but it looks like the class was already added to the `target-sim` folder, and we already have docs about the simulator implementation on our [docs](https://www.winglang.io/docs/standard-library/cloud/domain#target-specific-details). @marciocadev should this be removed?

Comment (Chriscbr) [reply libs/wingsdk/src/target-sim/domain.ts:28]: In any case it doesn't seem like `Domain` has any inflight methods, so I'm okay leaving in some of this boilerplate for now for

**Pull request merged**